Remotion/Data/DomainObjects/ConfigurationLoader/ReflectionBasedConfigurationLoader/LegacyPropertyDefaultValueProvider.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Linq;
+using Remotion.Data.DomainObjects.Mapping;
+using Remotion.ExtensibleEnums;
+using Remotion.Reflection;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.ConfigurationLoader.ReflectionBasedConfigurationLoader
+{
+  /// <summary>
+  /// Implements the legacy behavior for <see cref="PropertyDefinition.DefaultValue"/>, which sets non-nullable reference-<see cref="Type"/> properties to
+  /// non-<see langword="null"/> values. This affects all <see cref="Array"/>, <see cref="string"/>, and <see cref="ExtensibleEnum{T}"/> properties.
+  /// </summary>
+  public class LegacyPropertyDefaultValueProvider : IPropertyDefaultValueProvider
+  {
+    public object? GetDefaultValue (IPropertyInformation propertyInfo, bool isNullable)
+    {
+      if (isNullable)
+        return null;
+
+      var propertyType = propertyInfo.PropertyType;
+      if (propertyType.IsArray)
+      {
+        var elementType = propertyType.GetElementType();
+        Assertion.DebugIsNotNull(elementType, "elementType != null when _propertyType.IsArray");
+        return Array.CreateInstance(elementType, 0);
+      }
+
+      if (propertyType == typeof(string))
+        return string.Empty;
+
+      if (propertyType.IsEnum)
+      {
+        var firstValueOrNull = EnumUtility.GetEnumMetadata(propertyType).OrderedValues.FirstOrDefault();
+        if (firstValueOrNull == null)
+          firstValueOrNull = Activator.CreateInstance(propertyType);
+
+        return firstValueOrNull;
+      }
+
+      if (ExtensibleEnumUtility.IsExtensibleEnumType(propertyType))
+      {
+        // TODO if there are no values, should be return a dummy value?
```

Comment:
We have to decide whether to return `null ` or make up a value. The previous implementation of `PropertyDefinition.DefaultValue` threw an exception, which we should not do (see comment on https://github.com/re-motion/Framework/pull/674/files/69a0a3e16236e5b720a96313306a1bcdfbd70cf1#diff-758b70ed37a08eceb6a718ee93ead95d66319b5e419730933d29554173d4b27d)

While it is possible to return `null` here, it seems more consistent with the legacy behavior to create a dummy value. In the grand scheme of things it should not matter because the mapping validation will throw an error.

Reply:
Good. So, then the decision is to use a dummy value based on the IExtensibleEnum interface. I think we have a name suggestion for that somewhere in the ticket?

Reply:
Ah, because we have a typeguard in there. I.C. 

>We could use FormatterServices.GetSafeUninitializedObject(Type type) like we do for DomainObjects, since ExtensibleEnum<T> is [Serializable], and then use PopulateObjectMembers() to set ValueName to "Uninitialized".
Sounds good to me.

> Or we could have a repository of those objects within the LegacyPropertyDefaultValueProvider, because we only intend to use it there.
I prefer that one.